tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/rawtemplate/RawTemplateLiteralTest.java

```diff
@@ -118,6 +122,13 @@ public void testIndirectAssignmentToConcreteType() {
         BRunUtil.invoke(result, "testIndirectAssignmentToConcreteType");
     }
 
+    @Test(expectedExceptions = BLangRuntimeException.class,
+          expectedExceptionsMessageRegExp =
+                  ".*InvalidUpdate message=modification not allowed on readonly value.*")
```

Comment:
Shall we test this also in Ballerina itself?

For example, https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-Preview1/tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field.bal#L149.